tests/integration/golang/mlflow/artifact/get_artifact_local_test.go

```diff
@@ -0,0 +1,256 @@
+			runID := strings.ReplaceAll(uuid.New().String(), "-", "")
+			runArtifactDir := filepath.Join(experimentArtifactDir, runID, "artifacts")
+			run, err := s.runFixtures.CreateRun(context.Background(), &models.Run{
+				ID:             runID,
+				Status:         models.StatusRunning,
+				SourceType:     "JOB",
+				ExperimentID:   *experiment.ID,
+				ArtifactURI:    fmt.Sprintf("%s%s", tt.prefix, runArtifactDir),
+				LifecycleStage: models.LifecycleStageActive,
+			})
+			assert.Nil(s.T(), err)
+
+			// 3. create artifacts.
+			err = os.MkdirAll(runArtifactDir, fs.ModePerm)
+			assert.Nil(s.T(), err)
+			err = os.WriteFile(filepath.Join(runArtifactDir, "artifact.file1"), []byte("contentX"), fs.ModePerm)
+			assert.Nil(s.T(), err)
+			err = os.Mkdir(filepath.Join(runArtifactDir, "artifact.dir"), fs.ModePerm)
+			assert.Nil(s.T(), err)
+			err = os.WriteFile(filepath.Join(runArtifactDir, "artifact.dir", "artifact.file2"), []byte("contentXX"), fs.ModePerm)
+			assert.Nil(s.T(), err)
+
+			// 4. make actual API call for root dir file
+			rootFileQuery, err := urlquery.Marshal(request.GetArtifactRequest{
+				RunID: run.ID,
+				Path:  "artifact.file1",
+			})
+			assert.Nil(s.T(), err)
+
+			resp, err := s.serviceClient.DoStreamRequest(
+				http.MethodGet,
+				fmt.Sprintf("%s%s?%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute, rootFileQuery),
+				nil,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "contentX", string(resp))
+
+			// 5. make actual API call for sub dir file
+			subDirQuery, err := urlquery.Marshal(request.GetArtifactRequest{
+				RunID: run.ID,
+				Path:  "artifact.dir/artifact.file2",
+			})
+			assert.Nil(s.T(), err)
+
+			resp, err = s.serviceClient.DoStreamRequest(
+				http.MethodGet,
+				fmt.Sprintf("%s%s?%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute, subDirQuery),
+				nil,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "contentXX", string(resp))
+		})
+	}
+}
+
+func (s *GetArtifactLocalTestSuite) Test_Error() {
+	// create test experiment
+	experimentArtifactDir := s.T().TempDir()
+	experiment, err := s.experimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+		Name:             fmt.Sprintf("Test Experiment In Path %s", experimentArtifactDir),
+		LifecycleStage:   models.LifecycleStageActive,
+		ArtifactLocation: experimentArtifactDir,
+	})
+	assert.Nil(s.T(), err)
+
+	// create test run
+	runID := strings.ReplaceAll(uuid.New().String(), "-", "")
+	runArtifactDir := filepath.Join(experimentArtifactDir, runID, "artifacts")
+	_, err = s.runFixtures.CreateRun(context.Background(), &models.Run{
+		ID:             runID,
+		Status:         models.StatusRunning,
+		SourceType:     "JOB",
+		ExperimentID:   *experiment.ID,
+		ArtifactURI:    runArtifactDir,
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+
+	err = os.MkdirAll(filepath.Join(runArtifactDir, "subdir"), fs.ModePerm)
+	assert.Nil(s.T(), err)
+
+	testData := []struct {
```

Comment:
sometimes we use `testCases` sometimes `testData`, can you please unity it? for me `tests` just works really well.

Reply:
sure